pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -222,8 +222,13 @@ async def add_tool(tool: Tool[DepsT]) -> None:
     async def add_mcp_server_tools(server: MCPServer) -> None:
         if not server.is_running:
             raise exceptions.UserError(f'MCP server is not running: {server}')
-        tool_defs = await server.list_tools()
-        # TODO(Marcelo): We should check if the tool names are unique. If not, we should raise an error.
+        tool_defs: list[ToolDefinition] = await server.list_tools()
+
+        tool_names: list[str] = [tool_def.name for tool_def in tool_defs]
+
+        if len(set(tool_names)) != len(tool_names):
+            raise exceptions.UserError(f'Tool names must be unique: {str(tool_names)}')
```

Comment:
I think it would be a more pleasant experience if we can point out from which MCP server and which tool has duplicated name in the exception.